ValidationLibrary.AzureFunctions.Tests/StatusEndpointTests.cs

```diff
@@ -0,0 +1,54 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using NSubstitute;
+using NUnit.Framework;
+using Octokit;
+using ValidationLibrary.Rules;
+
+namespace ValidationLibrary.AzureFunctions.Tests
+{
+    [TestFixture]
+    public class StatusEndpointTests
+    {
+        private StatusEndpoint _statusEndpoint;
+
+        [Test]
+        public void Run_NormalStatusEndpointCheck()
+        {
+            Environment.SetEnvironmentVariable("GitHub:Organization", "mock");
+            Environment.SetEnvironmentVariable("GitHub:Token", "mock");
+            
+            var ruleType = typeof(HasLicenseRule);
+            var expectedRulesTypes = ruleType.Assembly.GetExportedTypes().Where(t => t.GetInterface(nameof(IValidationRule)) != null && !t.IsAbstract);
+            var expectedRules = expectedRulesTypes.Select(r =>
+            {
+                var args = r.GetConstructors()[0].GetParameters().Select(p => (object)null).ToArray();
+                return ((IValidationRule)Activator.CreateInstance(r, args));
+            }).ToArray();
+            var expectedStatuses = expectedRules.Select(r => r.GetConfiguration());
+
+            var validator = Substitute.For<ValidationLibrary.RepositoryValidator>(new object[] { Substitute.For<ILogger<ValidationLibrary.RepositoryValidator>>(), Substitute.For<IGitHubClient>(), expectedRules });
```

Comment:
This is the tricky part there, but we are missing a piece (You didn't miss it, it just doesn't exist yet. We should create `IRepositoryValidator` in `ValidationLibrary` and make `ValidationLibrary.RepostioryValidator` use it. Then we can just `Substitute.For<IRepositoryValidator>()` with less of a hassle.

Reply:
With the 
> var rule = Substitute.For<IValidationRule>();
> rule.GetConfiguration().Returns(new Dictionary<string,string>());
bit the test is much more focused and not so much need for this but it may be good for other reasons too. Anyway, this is the last commit and I can reverse it easily